packages/firebase_admob/README.md

```diff
@@ -182,6 +182,95 @@ RewardedVideoAd.instance.listener =
 Because `RewardedVideoAd` is a singleton object, it does not offer a `dispose`
 method.
 
+## Using native ads
+
+Native ads are presented to users via UI components that
+are native to the platform. (e.g. A
+[View](https://developer.android.com/reference/android/view/View) on Android or a
+[UIView](https://developer.apple.com/documentation/uikit/uiview?language=objc)
+on iOS). Using Flutter widgets to create native ads is NOT supported by
+this.
+
+### Android
+The Android Admob Plugin requires a class that implements `NativeAdFactory` which calls `createNativeAd(
+[UnifiedNativeAd](https://developers.google.com/android/reference/com/google/android/gms/ads/formats/UnifiedNativeAd) nativeAd,
+Map<String, Options> customOptions)` and returns a
+[UnifiedNativeAdView](https://developers.google.com/android/reference/com/google/android/gms/ads/formats/UnifiedNativeAdView).
+
+You can implement this in your `MainActivity.java` or create a separate class in the same directory
+as `MainActivity.java` as seen below:
+
+```java
+package my.app.path;
+
+import com.google.android.gms.ads.formats.UnifiedNativeAd;
+import com.google.android.gms.ads.formats.UnifiedNativeAdView;
+import io.flutter.plugins.firebaseadmob.FirebaseAdMobPlugin.NativeAdFactory;
+import java.util.Map;
+
+class NativeAdFactoryExample implements NativeAdFactory {
+  @Override
+  public UnifiedNativeAdView createNativeAd(
+      UnifiedNativeAd nativeAd, Map<String, Object> customOptions) {
+    // Create UnifiedNativeAdView
+  }
+}
+```
+ 
+After, pass an instance of `NativeAdFactory` to the `FirebaseAdMobPlugin`. This is done differently whether
+you are using the Embedding V1 or Embedding V2. If you're using the Embedding V1, you're `MainActivity.java`
+should look similar to:
+
+```java
+package my.app.path;
+
+import android.os.Bundle;
+import io.flutter.app.FlutterActivity;
+import io.flutter.plugins.GeneratedPluginRegistrant;
+import io.flutter.plugins.firebaseadmob.FirebaseAdMobPlugin;
+
+public class MainActivity extends FlutterActivity {
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    GeneratedPluginRegistrant.registerWith(this);
+
+    final FirebaseAdMobPlugin adMobPlugin =
+        valuePublishedByPlugin("io.flutter.plugins.firebaseadmob.FirebaseAdMobPlugin");
```

Comment:
How about making a static method on FirebaseAdMobPlugin to take care of the registration?

So this would look like:
```java
FirebaseAdMobPlugin.registerNativeAdFactory(this, <potential other params>);
```

We could overload this method for the v2 use case as well, so for V2 it looks like:

```java
FirebaseAdMobPlugin.registerNativeAdFactory(flutterEngine, <potential other params>);
```

This has the benefit of keeping the implementation details of how to fetch the plugin (e.g what is the plugin's class name) as part of the plugin's implementation.

Reply:
I assumed that the move to the v2 embedding included avoiding the model of static plugins. I assumed we wanted to make everything about a plugin to be an instance. In an add-to-app scenario, what if we have multiple `FlutterView`s that want to use their own `NativeAdFactory`s.

Reply:
Not sure I follow, the v2 version of this method registers for a given `FlutterEngine`... if you have 2 FlutterEngine you could still register a separate factory for each engine...

Reply:
Oh, were u suggesting that we have a static method only for the v1 embedding?

Reply:
clarified f2f. done